update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+            in_cert_part = False
+            cert_start_reg = re.compile("^'cert' \?=")
+            cert_end_reg = re.compile("^EOF")
+            for line in template.readlines():
+                if in_cert_part:
+                    if re.search(cert_end_reg, line):
+                        in_cert_part = False
+                        break
+                    else:
+                        old_cert += line
+                elif re.search(cert_start_reg, line):
+                    in_cert_part = True
+            template.close()
+            if old_cert:
+                try:
+                    old_x509_cert = crypto.load_certificate(
+                        crypto.FILETYPE_PEM, old_cert
+                    )
+                except crypto.Error as e:
+                    sys.stderr.write("ERROR Existing certificate for " +
+                                     "%s could not be " % hostname +
+                                     "loaded: %s\n" % str(e))
+                    old_x509_cert = None
+                if old_x509_cert is not None:
+                    if old_x509_cert.has_expired():
+                        sys.stdout.write("Existing certificate" +
+                                         "(%s) is no " % server_cert_tpl +
+                                         "longer valid: ignoring it.\n")
+                        old_x509_cert = None
+                    elif (old_x509_cert.digest('SHA256')
+                          == x509_cert.digest('SHA256')):
+                        sys.stdout.write("Existing certificate " +
+                                         "(%s) is the " % server_cert_tpl +
+                                         "same than the retrieved " +
+                                         "certificate: ignoring it.\n")
+                        old_x509_cert = None
+        sys.stdout.write("Updating template for VOMS server " +
+                         "%s (%s)\n" % (hostname, server_cert_tpl))
+        template = open(server_cert_tpl, 'w')
+        template.write("structure template %s;\n\n" % server_cert_ns)
+        template.write("'cert' ?= <<EOF;\n")
+        if x509_cert:
+            template.write(crypto.dump_certificate(
+                crypto.FILETYPE_PEM, x509_cert
+            ))
+        else:
+            template.write("\n")
+        template.write("EOF\n\n")
+        if old_x509_cert:
+            sys.stdout.write("Existing certificate ('oldcert') found and " +
+                             "different from VO ID card: 'oldcert' defined\n")
+            template.write("'oldcert' ?= <<EOF;\n")
+            template.write(
+                crypto.dump_certificate(crypto.FILETYPE_PEM, old_x509_cert)
+            )
+            template.write("EOF\n\n")
+        template.close()
+
+
+def write_vo_list(vo_details, basedir, namespaces):
+    """Write template containing a list of all defined VOs
+    """
+    vo_list_ns = namespaces['params_dir'] + os.path.sep + 'allvos'
```

Comment:
I believe `os.path.join()` is preferred.

Reply:
The code has been modified to use ```os.path.join()```.